tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -3105,6 +3107,72 @@ int BPF_KPROBE(trace_mprotect_alert)
     return 0;
 }
 
+#define BPF_OBJ_NAME_LEN 16U
+
+SEC("kprobe/security_bpf")
+int BPF_KPROBE(trace_security_bpf)
+{
+    if (!should_trace())
+        return 0;
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    context_t context = init_and_save_context(ctx, submit_p, SECURITY_BPF, 1 /*argnum*/, 0 /*ret*/);
+
+    u64 *tags = bpf_map_lookup_elem(&params_names_map, &context.eventid);
+    if (!tags)
+        return -1;
+
+    /* 1st argument == map_cmd (int) */
+    int map_cmd = (int)PT_REGS_PARM1(ctx);
+
+    save_to_submit_buf(submit_p, (void *)&map_cmd, sizeof(int), INT_T, DEC_ARG(0, *tags));
+
+    events_perf_submit(ctx);
+    return 0;
+};
+
+SEC("kprobe/security_bpf_map")
+int BPF_KPROBE(trace_security_bpf_map)
+{
+    if (!should_trace())
+        return 0;
+
+    buf_t *submit_p = get_buf(SUBMIT_BUF_IDX);
+    if (submit_p == NULL)
+        return 0;
+
+    set_buf_off(SUBMIT_BUF_IDX, sizeof(context_t));
+
+    context_t context = init_and_save_context(ctx, submit_p, SECURITY_BPF_MAP, 0 /*argnum*/, 0 /*ret*/);
+
+    u64 *tags = bpf_map_lookup_elem(&params_names_map, &context.eventid);
+    if (!tags)
+        return -1;
+
+    struct bpf_map *map;
+    char map_name[BPF_OBJ_NAME_LEN];
+
+    map = (struct bpf_map *)PT_REGS_PARM1(ctx);
+    bpf_probe_read_kernel_str(&map_name, BPF_OBJ_NAME_LEN, (void *)&map->name);
+
+    if (*map_name != '\0') {
+        context.argnum+=2;
+        save_context_to_buf(submit_p, (void*)&context);
```

Comment:
You can set argnum to 2 once in `init_and_save_context` above.
We update the context's argnum when there is more than one possibility for the number of arguments, for example, when (and if) we will have the "tainted" boolean

Reply:
Yep, let's just pretend I did not do this heh. Will fix.

Reply:
Yep, does not like we need the:

```
if (*map_name != '\0')
```

any longer. Fixing this and the argnum.